backend/app.py

```diff
@@ -157,6 +161,92 @@ def discord_callback():
 
 
+@app.route("/admin/add_points", methods=['POST'])
+def add_points():
+    """
+    Add points
+    """
+    data = request.json
+
+    amount = data['amount']
+    assignee = data['assignee']
+    description = data['description']
+    event_id = None
+
+    # if user's discord id is given, change assignee to discord username
+    if "#" in assignee:
+        user = User.query.filter_by(name=assignee).first()
+    else:
+        user = User.query.filter_by(id=assignee).first()
+        assignee = user.name
+
+    discord_id = user.id
+
+    if description == 'Event':
+        event_id = data.get('event_id')
+        secret_input = data.get('secret_input')
+        if event_id is None:
+            return jsonify({
+                "success": False,
+                "message": 'Please specify the event id'
+            })
+        if secret_input is None:
+            return jsonify({
+                "success": False,
+                "message": 'Please input the secret code'
+            })
+
+        # Check if points are already claimed for event
+        if Points.query.filter_by(event_id=event_id, assignee=discord_id).first():
+            return jsonify({
+                "success": False,
+                "message": 'Event points already claimed'
+            })
+        else:
+            # Check if input matches event secret code
+            event = Event.query.filter_by(id=event_id).first()
+            if event.secret_code == secret_input:
+                amount = event.points_amount
+                message = f'{amount} points added to {assignee} for Event {event.name}'
+                success = True
+            else:
+                return jsonify({
+                    "success": False,
+                    "message": f'The code {secret_input} is incorrect for Event {event.name}'
+                })
+
+    elif description == 'Discord':
+        # Check daily limit of 5 messages is exceeded
+        discord_points_today = Points.query.filter_by(description='Discord', assignee=discord_id) \
+            .filter(func.date(Points.timestamp) == func.date(func.now())).all()
+        if len(discord_points_today) >= 5:
+            return jsonify({
+                "success": False,
+                "message": 'Daily limit for Discord activity points reached'
+            })
+        else:
+            message = f'{amount } points added to {assignee} for Discord activity'
+            success = True
+    else:
+        message = f'{amount} points added to {assignee} for {description}'
+        success = True
+
+    # Create a Points in the points table
+    new_point = Points(amount=amount, assignee=discord_id,
+                       description=description, event_id=event_id)
+    db.session.add(new_point)
+
+    # Add to user's total points
+    user.points_total += amount
+
+    db.session.commit()
+
+    return jsonify({
+        "success": success,
+        "message": message
+    })
```

Comment:
Could you send back the amount (amount of points added), discord_id/handle, and timestamp as JSON. Really need it for when the description is an 'Event' and the client needs to know how many points were added.
Maybe something like this:
```json
{
    "success": success,
    "message": message,
    "data": {
        "amount": amount,
        "assignee": assignee,
        "timestamp": ...
    }
}
```

Reply:
Added @PawanKolhe